build/maven.go

```diff
@@ -174,43 +175,72 @@ func (mm *MavenModule) loadMavenHome() (mavenHome string, err error) {
 		// We need to grab the location using the mvn --version command
-
 		// First we will try lo look for 'mvn' in PATH.
-		mvnPath, err := exec.LookPath("mvn")
-		if err != nil || mvnPath == "" {
-			return "", errors.New(err.Error() + "Hint: The mvn command may not be included in the PATH. Either add it to the path, or set the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
-		}
-		mm.containingBuild.logger.Debug(MavenHome, " is not defined. Retrieving Maven home using 'mvn --version' command.")
-		cmd := exec.Command("mvn", "--version")
-		var stdout bytes.Buffer
-		cmd.Stdout = &stdout
-		err = cmd.Run()
+		maven, err := mm.getExecutableName()
 		if err != nil {
-			return "", err
+			return maven, err
 		}
-		output := strings.Split(strings.TrimSpace(stdout.String()), "\n")
-		// Finding the relevant "Maven home" line in command response.
-		for _, line := range output {
-			if strings.HasPrefix(line, "Maven home:") {
-				mavenHome = strings.Split(line, " ")[2]
-				if runtime.GOOS == "windows" {
-					mavenHome = strings.TrimSuffix(mavenHome, "\r")
-				}
-				mavenHome, err = filepath.Abs(mavenHome)
-				if err != nil {
-					return "", err
-				}
-				break
+		if !mm.extractorDetails.useWrapper {
+			mvnPath, err := exec.LookPath("mvn")
+			if err != nil || mvnPath == "" {
+				return "", errors.New(err.Error() + "\nHint: The mvn command may not be included in the PATH. Either add it to the path or set the M2_HOME environment variable value to the maven installation directory, which is the directory that includes the bin and lib directories.")
 			}
 		}
-		if mavenHome == "" {
+		output, stdout, err := mm.getMavenVersionOutPut(maven)
+		if err != nil {
+			return "", errors.New("Could not find the location of the maven home directory, by running 'mvn --version' command. The command output is:\n" + stdout.String() + "\nYou also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
+		}
+		// Finding the relevant "Maven home" line in command response.
+		mavenHome, err = mm.ExtractMavenPath(output)
+		if mavenHome == "" || err != nil {
 			return "", errors.New("Could not find the location of the maven home directory, by running 'mvn --version' command. The command output is:\n" + stdout.String() + "\nYou also have the option of setting the M2_HOME environment variable value to the maven installation directory, which is the directory which includes the bin and lib directories.")
 		}
 	}
 	mm.containingBuild.logger.Debug("Maven home location: ", mavenHome)
+
 	return
 }
 
+func (mm *MavenModule) getExecutableName() (maven string, err error) {
+	maven = "mvn"
+	if mm.extractorDetails.useWrapper {
+		maven = "./mvnw"
+		if utils.IsWindows() {
+			maven, err = filepath.Abs("mvnw.cmd")
```

Comment:
Can we also return the absolute path for non-Windows systems?

Reply:
sure, but we will also need the if{} because we have to add ".cmd" when looking the path in windows

Reply:
Done